ci/check-style.sh

```diff
@@ -27,30 +27,52 @@ readonly BINDIR="$(dirname "$0")"
 
 # Build paths to ignore in find(1) commands by reading .gitignore.
 declare -a ignore=( -path ./.git )
-if [ -f .gitignore ]; then
+if [[ -f .gitignore ]]; then
   while read -r line; do
     case "${line}" in
-    [^#]*/*) ignore+=( -o -path "./$(expr "${line}" : '\(.*\)/')" );;
-    [^#]*)   ignore+=( -o -name "${line}" );;
+    [^#]*/*) ignore+=( -o -path "./$(expr "${line}" : '\(.*\)/')" ) ;;
+    [^#]*)   ignore+=( -o -name "${line}" ) ;;
     esac
   done < .gitignore
 fi
 
 find google/cloud -name '*.h' -print0 |
   xargs -0 awk -f "${BINDIR}/check-include-guards.gawk"
 
+replace_original_if_changed() {
+  if [[ $# != 1 ]]; then
+    return 1
+  fi
+
+  local file="$1"
+  shift
+
+  chmod --reference="${file}" "${file}.tmp"
+  if cmp -s "${file}" "${file}.tmp"; then
+    rm -f "${file}.tmp"
+  else
+    mv -f "${file}.tmp" "${file}"
+  fi
+}
+
 # Apply cmake_format to all the CMake list files.
 #     https://github.com/cheshirekow/cmake_format
 find . \( "${ignore[@]}" \) -prune -o \
        \( -name 'CMakeLists.txt' -o -name '*.cmake' \) \
        -print0 |
-  xargs -0 cmake-format -i
+  while IFS= read -r -d $'\0' file; do
+    cmake-format "${file}" >"${file}.tmp"
+    replace_original_if_changed "${file}"
```

Comment:
It's a little confusing that `"${file}.tmp"` isn't mentioned here.  I think I'd just make the function take two names (i.e., not assume `.tmp`).